backend/test/infra/controller/TransactionController.test.ts

```diff
@@ -153,4 +158,116 @@ describe("TransactionController", () => {
       });
     });
   });
+
+  describe("Remove transaction", () => {
+    let server: FastifyInstance;
+    const groupRepo = new GroupRepositoryMock();
+    const removeTransactionToGroupMock = {
+      execute: vi.fn<
+        RemoveTransactionToGroupRequestObject,
+        RemoveTransactionToGroupResponseObject
+      >(),
+    };
+
+    beforeEach(async () => {
+      const module = await import("src/core/usecase/RemoveTransactionToGroup");
+      // @ts-expect-error It isn't typed as a Mock
+      module.RemoveTransactionToGroup.mockReturnValue(
+        removeTransactionToGroupMock
+      );
+
+      // server = await createServer("test");
+      server = Fastify({ logger: false });
+      await server.register(TransactionController, {
+        prefix: `/group/:groupId/transaction`,
+        repositories: { group: groupRepo },
+      });
+      await server.ready();
+    });
+
+    afterEach(() => {
+      vi.clearAllMocks();
+      vi.resetAllMocks();
+    });
+
+    test("Given a transaction to remove, it should succeed with a status 201 and return the updated group", async () => {
+      // Arrange
+      removeTransactionToGroupMock.execute.mockResolvedValue({
+        success: true,
+        payload: {
+          id: 0,
+          name: "new group",
+          members: [new GroupMember(0, "John")],
+          transactions: [],
+        },
+      });
+
+      // Act
+      const response = await server.inject({
+        method: "DELETE",
+        url: `/group/${0}/transaction`,
+        payload: { transactionId: 0 },
+      });
+
+      // Assert
+      expect(removeTransactionToGroupMock.execute).toBeCalledWith(0, 0);
+      expect(response.statusCode).toBe(200);
+      expect(response.json()).toEqual({
+        id: 0,
+        name: "new group",
+        members: [new GroupMember(0, "John")],
+        transactions: [],
+      });
+    });
+
+    test("Given no payload, it should return an error 400", async () => {
+      const response = await server.inject({
+        method: "DELETE",
+        url: `/group/${0}/transaction`,
+      });
+
+      expect(response.statusCode).toBe(400);
+    });
+
+    test("Given an empty payload, it should return an error 400", async () => {
+      const response = await server.inject({
+        method: "DELETE",
+        url: `/group/${0}/transaction`,
+        payload: {},
+      });
+
+      expect(response.statusCode).toBe(400);
+    });
+
+    test("Given not a full payload, it should return an error 400", async () => {
+      const response = await server.inject({
+        method: "DELETE",
+        url: `/group/${0}/transaction`,
+        payload: { groupId: 0, payerId: 0 },
+      });
+
+      expect(response.statusCode).toBe(400);
+    });
```

Comment:
Strange payload for the delete (even if in the end, this test won't be necessary)